### PR TITLE
exempt processlist column renaming through aliases

### PR DESCRIPTION
**Pull request by jycor**
needed for dolphie to work; extension of https://github.com/dolthub/go-mysql-server/pull/2764

### CODE REVIEW

File: enginetest/enginetests.go [hunk 1]
```diff
@@ -359,6 +359,95 @@ func TestInfoSchema(t *testing.T, h Harness) {
 		)
 	})
 
+	t.Run("information_schema.processlist projection with alias case", func(t *testing.T) {
```
Comment (max-hoffman) [inline enginetest/enginetests.go:362]: fairly painful test setup, bit late in the day for better ideas thoguh

File: sql/planbuilder/scalar.go [hunk 2]
```diff
@@ -130,7 +131,24 @@ func (b *Builder) buildScalar(inScope *scope, e ast.Expr) (ex sql.Expression) {
 			}
 			b.handleErr(err)
 		}
-		c = c.withOriginal(v.Name.String())
+		// Look past table aliases
+		var origTbl string
+		if inScope.node != nil {
```
[the remark this answers is not in the document]
Comment (max-hoffman) [reply sql/planbuilder/scalar.go:136]: and maybe put in a helper func

**Review by max-hoffman — APPROVED** [review]
mostly good, if this is a perf-sensitive path might need to figure out something else

File: sql/planbuilder/scalar.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jycor) [reply sql/planbuilder/scalar.go]: `inScope.tables` doesn't have info on whether or not this table is being aliased
added a helper func

**Pull request merged**